Fix Config.get_terminal_writer() crash when terminalreporter is unregistered (#14378)
pytest-dev/pytest: changelog/14377.bugfix.rst

> Fixed :meth:`pytest.Config.get_terminal_writer` crashing with an internal ``AssertionError`` when the terminal reporter plugin has been unregistered (for example, by :pypi:`pytest-tap` in streaming mode). A fresh terminal writer is now created on demand.

Reviewer: The `get_terminal_writer` function is private, so you can't ref it. Also, let's remove the last sentence describing the solution since that's not relevant for a changelog.

```suggestion
Fixed a crash from `Config.get_terminal_writer` when the terminalreporter plugin has been unregistered (for example, by :pypi:`pytest-tap` in streaming mode).
```

Author: Thanks @bluetech.

>I would rather look at removing the dependency on TerminalWriter from assertrepr_compare.

I think you're right. I was proposing a solution, mostly to highlight the problem and in case the solution would be acceptable (after all... it's crashing without the change, and it's not crashing with the change).

>Let me know if you want to try it, if not, I'll try looking into it myself.

I think you have more knowledge of the code base and will do a better job with less effort. So if you're OK, I'd leave that one to you. We can close the PR whenever you want.

Or I may try to tackle it in a few months if I see it's still not been touched (which would be totally understandable).

Thank you for your time.
